src/ast/mod.rs

```diff
@@ -4050,6 +4050,14 @@ pub enum Statement {
         arguments: Vec<Expr>,
         options: Vec<RaisErrorOption>,
     },
+    /// Go (SQL Server)
+    ///
+    /// GO is not a Transact-SQL statement; it is a command recognized by various tools as a batch delimiter
+    ///
+    /// See: https://learn.microsoft.com/en-us/sql/t-sql/language-elements/sql-server-utilities-statements-go
+    Go {
+        count: Option<u64>,
+    },
```

Comment:
Repr wise can we wrap the statement body in explicit structs? we're[ transitioning away](https://github.com/apache/datafusion-sqlparser-rs/issues/1204) from anonymous structs in order to make the API more ergonomic. Something like
```rust
struct GoStatement {
    count: Option<u64>
}
Statement::Go(GoStatement)
```

Reply:
Done 👍